src/service/resolver/amazon/amazon-resolver.service.ts

```diff
@@ -35,19 +38,113 @@ export class AmazonResolverService extends AbstractResolver {
   private static readonly KINDLE = 'kindle';
 
   private static readonly URL_PREFIX = '/dp/';
+  private static readonly AMAZON_HOSTS: string[] = [
```

Comment:
Move this logic to a new service and track the number of redirects, to stop it after a count.
Add stats to track redirects.